### PR TITLE
selinux: create Vagrantfile for SELinux

### PR DESCRIPTION
**Pull request by Gekko0114**
In this PR, created Vagrantfile for an SELinux environment.

ref: https://github.com/containers/youki/pull/2850

### CODE REVIEW

File: experiment/selinux/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure("2") do |config|
+    config.vm.box = "centos/8"
+    config.vm.synced_folder '.', '/vagrant/youki', disabled: false
+
+    config.vm.provider "virtualbox" do |v|
```
Comment (Gekko0114) [inline experiment/selinux/Vagrantfile:8]: I couldn't use VirtualBox on my PC, so I was using libvirt instead.
I haven't tried this Vagrantfile (After checking this vagrantfile works for libvirt, updated for virtualbox)

**Comment by Gekko0114** [comment]
Hi @utam0k @YJDoc2 
I've created a PR creating Vagrantfile for SELinux. Could you review it?

**Comment by YJDoc2** [comment]
Hey, I'll take a look and also validate the file works with virtualbox, as that's what I use.

File: experiment/selinux/Vagrantfile (hunk 1 above)
Comment (YJDoc2) [reply experiment/selinux/Vagrantfile:8]: Hey, can you try the image `generic/centos8` instead? For some reason the `centos/8` box has issues mounting the youki folder on my machine, while the generic one works.

Comment (Gekko0114) [reply experiment/selinux/Vagrantfile:8]: Hi @YJDoc2 
`generic/centos8` works on my env (libvirt) as well.
Thanks for your investigation.

**Comment by Gekko0114** [comment]
Hi @YJDoc2 
I've fixed your comment. Could you review it again?

**Comment by YJDoc2** [comment]
Hey, so the vagrant box is working for me now. A couple of things -
1. We can remove the `git clone` command. We are already mounting the current directory in the box, and we would need that to do development, so no need to clone the repo.
2. How do I validate that SELinux is enabled and working as expected in the box? On my host, where I know selinux is disabled, I get from `cargo run`
```
selinux is not enabled
Failed to set enforce mode of SELinux: can't get selinux enforce path
default enforce mode is: disabled
current enforce mode is: disabled
Invalid format for SELinux label: unconfined

Error: Failed to set file label for SELinux: Failed to set_xattr: No data available (os error 61)
```

In the vagrant box I am getting 
```
selinux is enabled
default enforce mode is: enforcing
current enforce mode is: enforcing
SELinux label of current process is: unconfined_u:unconfined_r:unconfined_t:s0-s0
Error: Failed to set file label for SELinux: Failed to set_xattr: Operation not supported (os error 95)
```

Is the last error ok? Is that issue with box setup or the demo code we have in the experimental crate?

Apart from this, it looks good. Thanks!

**Comment by Gekko0114** [comment]
> Is the last error ok? Is that issue with box setup or the demo code we have in the experimental crate?

I've updated main.rs in this PR, so it has been fixed now.

> We can remove the git clone command. We are already mounting the current directory in the box, and we would need that to do development, so no need to clone the repo.

Sure, then I will remove git clone.

**Comment by YJDoc2** [comment]
> I've updated main.rs in this PR, so it has been fixed now.

Ok, great. Can I ask you to either rebase or do main merge?

> Sure, then I will remove git clone.

Thanks!

Apart from these it looks good, so once you update, I'll approve and merge.

**Comment by Gekko0114** [comment]
> Can I ask you to either rebase or do main merge?

Sure, then I will rebase and force push this PR later today

**Comment by Gekko0114** [comment]
Hi @YJDoc2 
I've removed git command and rebased my branch. Could you review it ?

**Comment by YJDoc2** [comment]
Hey @Gekko0114 Thanks for updating. I'm still receiving the 
```
Error: Failed to set file label for SELinux: Failed to set_xattr: Operation not supported (os error 95)
```
Error. Can this be an issue with my underlying system?
I am running vagrant up, then going into the `/vagrant/youki` dir and doing `cargo run`. Is that correct?

**Comment by Gekko0114** [comment]
Yes, it is correct. Did you run on this branch?

**Comment by YJDoc2** [comment]
Hmm, the issue was that with the directory mounting. When I copied the binary into `/tmp` it worked correctly as expected and sets the label. This is probably issue with mounting dirs into vagrant, not not with our code, so can ignore. Going ahead and merging this PR. Thanks a lot!

**Pull request merged**